### PR TITLE
Added Python code for CohenSutherLand Algorithm

### LINKED ISSUE #858: Cohen Sutherland Line Clipping Algorithm
I would like to add the Cohen Sutherland algorithm in C/C++

### PR DESCRIPTION
**Pull request by prekshajagetiya123**
Fixes #858 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Cohen Sutherland Line Clipping Algorithm 

#### Languages Used:
- Python

#### Files Added:
- Cohen_Sutherland.py

### CODE REVIEW

**Comment by jainaman224** [comment]
Please add input and output.

File: CohenSutherlandLineClipping/Cohen_Sutherland.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
```
Comment (jainaman224) [inline CohenSutherlandLineClipping/Cohen_Sutherland.py]: Fix folder name. Please change it to `Cohen_Sutherland` as well.

**Comment by prekshajagetiya123** [comment]
I made the required changes suggested by you. Please review it!

File: Cohen_Sutherland/Cohen_Sutherland.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        code |= top 
+
+    return code 
+
+
+# Implemention of Cohen Sutherland Algorithm 
+# In order to clip a line from P to Q
+def CohenSutherlandClipping(x1, y1, x2, y2): 
+
+	# find region codes for P, Q
```
Comment (jainaman224) [inline Cohen_Sutherland/Cohen_Sutherland.py]: Please use 4 space indentation throughout the file.

File: Cohen_Sutherland/Cohen_Sutherland.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+# Sample Input 2
+# CohenSutherlandClipping(4,6, 7, 8)
+# Output 2
+# The line is rejected
+
+
+# Plotting the graph
+plt.plot(x1,x2, y1, y2) 
+plt.show()
```
Comment (jainaman224) [inline Cohen_Sutherland/Cohen_Sutherland.py]: Please upload image to some source and link here as output.

Comment (prekshajagetiya123) [reply Cohen_Sutherland/Cohen_Sutherland.py]: I uploaded the link to the output image, please review it.

**Comment by jainaman224** [comment]
Hey now there are 3 files for same. Please connect with us on slack.

**Comment by prekshajagetiya123** [comment]
I'm sorry for inconvenience caused.

**Comment by prekshajagetiya123** [comment]
Sir, I removed the duplicate files. Please review it.

**Comment by jainaman224** [comment]
Awesome.
Now use 4 space indentation instead of 8.

**Comment by prekshajagetiya123** [comment]
Sure

**Comment by prekshajagetiya123** [comment]
fixed

**Comment by prekshajagetiya123** [comment]
Fixed

File: Cohen_Sutherland/Cohen_Sutherland.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        code |= top
+
+    return code
+
+
+# Implemention of Cohen Sutherland Algorithm
+# In order to clip a line from P to Q
+def CohenSutherlandClipping(x1, y1, x2, y2):
+
+        # find region codes for P, Q
```
Comment (jainaman224) [inline Cohen_Sutherland/Cohen_Sutherland.py]: Fix indentation.

File: Cohen_Sutherland/Cohen_Sutherland.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    p = find_code(x1, y1)  	# find code for P
+    q = find_code(x2, y2)   # find code for Q
+    accept = 0
+
+    while True:
+        if p == 0 and q == 0:  # If both points are inside the rectangle
+            accept = 1
+            break
+
+            # If both points are outside rectangle
```
Comment (jainaman224) [inline Cohen_Sutherland/Cohen_Sutherland.py]: Fix Indentation with all comments.

**Comment by prekshajagetiya123** [comment]
done

**Comment by jainaman224** [comment]
Comments are not indented correctly.

**Comment by prekshajagetiya123** [comment]
Hey, I have indented the comments, please review it.

File: Cohen_Sutherland/Cohen_Sutherland.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+# At least one of the points is outside the rectangle,
+# therefore select it,
+            x = 1.0
+            y = 1.0
+            if p != 0:
+                point_out = p
+            else:
+                point_out = q
+
+# Find  the intersection point using formulas:
```
Comment (jainaman224) [inline Cohen_Sutherland/Cohen_Sutherland.py]: Have a look at them.

**Comment by prekshajagetiya123** [comment]
now I think I have fixed them. Please review it.

**Pull request merged**